src/main/java/teammates/logic/core/AccountsLogic.java

```diff
@@ -238,4 +242,42 @@ private void createStudentAccount(StudentAttributes student)
         accountsDb.createEntity(account);
     }
 
+    /**
+     * Updates the read notifications of an account.
+     */
+    public AccountAttributes updateReadNotifications(String googleId, String notificationId, Instant endTime)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        AccountAttributes a = accountsDb.getAccount(googleId);
+        if (a == null) {
+            throw new EntityDoesNotExistException("Invalid account id.");
+        }
+
+        if (endTime.isBefore(Instant.now())) {
+            throw new InvalidParametersException("Notification has expired.");
+        }
+        Map<String, Instant> updatedReadNotifications = a.getReadNotifications();
+
+        for (Map.Entry<String, Instant> notification : updatedReadNotifications.entrySet()) {
+            if (notification.getValue().isBefore(Instant.now())) {
+                updatedReadNotifications.remove(notification.getKey());
+            }
+        }
+
+        if (!notificationsLogic.doesNotificationExists(notificationId)) {
+            throw new EntityDoesNotExistException("Invalid notification id to read");
+        }
```

Comment:
```suggestion
        if (!notificationsLogic.doesNotificationExists(notificationId)) {
            throw new EntityDoesNotExistException("Trying to mark as read a notification that does not exist");
        }
```

This check could also be moved to the front of the method to throw the exception early if notification is not found.